### ISSUE TITLE
Compile errors not presented for Scala 2.13.1

### ISSUE THREAD

**Issue by davesmith00000** [scalameta/metals#1409]
**Describe the bug**
Compile errors are not reported/shown in my editor for most classes of compile error when compiling against Scala 2.13.1 specifically.
Compiling through SBT works correctly.
Switching back to 2.12.10 or 2.13.0 also works correctly in my editor.

**To Reproduce**
Steps to reproduce the behaviour:
1. Choose at random, a working for comprehension
2. Insert obvious nonsense like `_ <- fish`
3. VSCode and NeoVim report no error.
4. SBT compile reports `not found: value fish`

Other examples include duplicate class names in the same file or inserting a val pointing to something that doesn't exist, and anything else beyond simple syntax errors. It will correctly report, for example, things like `} expected but <- found` if you remove the "fish" in the steps above.

**Expected behaviour**
Expect the compile error `not found: value fish` to be shown/highlighted in my editor (VSCode or NeoVim).

**Installation:**
- Operating system: macOS (Sierra, for boring reasons...)
- Editor: Visual Studio Code, Vim
- Metals version: v0.8.0
- JDK - have tried both AdaptOpenJDK 8 and 11 - no difference
- Metals server - have also tried the latest SNAPSHOT version

**Comment by ckipp01** [comment]
Thanks for reporting @davesmith00000.

I tried your example and wasn't able to reproduce it with 2.13.1.

<img width="405" alt="Screenshot 2020-02-09 at 09 09 38" src="https://user-images.githubusercontent.com/13974112/74098776-2e872e00-4b1c-11ea-9c4b-9994e5eb49fb.png">

Just to make sure, does this only occur when switching back and between another version and 2.13.1, or if you start a brand new project and make a simple for comprehension do you also see it?

Also, while this is happening is anything interesting going on if you look in `.metals/metals.log`?

**Comment by davesmith00000** [comment]
Thanks @ckipp01. Does this help?

Project is a multi project build with 8 sub-projects cross compiling variously to JVM and JS. I'm attempting to do an upgrade to 2.13.1, not move back and forth. However, this looks to be project specific. I tried opening up a very small, standard issue project I'd never opened before in vscode (written by someone else) and it worked / errored correctly.

1. Close all editors
2. rm -fr .metals/
3. rm -fr .bloop/
4. rm project/metals.sbt
5. sbt clean
6. code .

Results after recompile:
1. Metals doctor is all green.
2. No errors reporting in the editor.

<img width="328" alt="vscode-no-error" src="https://user-images.githubusercontent.com/908709/74099480-2202d580-4b1c-11ea-9344-89bac17e0eb9.png">

3. SBT compile does error correctly.

<img width="335" alt="sbt-error" src="https://user-images.githubusercontent.com/908709/74099481-25965c80-4b1c-11ea-88c0-f2e5f38af8fd.png">

4. metals.log reports no semanticdb for the file with the error in it, but gives no explanation. So that looks important. :-)

[metals.log](https://github.com/scalameta/metals/files/4176647/metals.log)

**Comment by ckipp01** [comment]
Yea, without the semantic information for that file being available/created, I'm assuming a lot of stuff isn't working in that file for you. Is the repo public?

**Comment by davesmith00000** [comment]
Yeah basically nothing beyond the superficial is working. Find by symbol seems to be ok. It's a private repo at the moment, but I could let you in for a look? Do you have a gitlab account?

**Comment by ckipp01** [comment]
That'd be really helpful to try and reproduce. https://gitlab.com/ckipp01

**Comment by davesmith00000** [comment]
I've added you with guest access for now - I have to run out for a couple of hours apologies for the up coming radio silence. The last commit is the upgrade to 2.13.1, try the main project in the `indigo` directory. Thanks for taking a look, hope you manage to spot some obvious mistake I've made. :-D

**Comment by ckipp01** [comment]
Hey @davesmith00000, when I'm looking at the project view, I don't see any repos. Are you able to link me directly to it?

**Comment by davesmith00000** [comment]
Try this? https://gitlab.com/purple-kingdom-games/indigo
If it's easier to avoid the chat noise on here, you can DM me on twitter? https://twitter.com/davidjamessmith
If we get to a resolution we can post the answer back on here?

**Comment by davesmith00000** [comment]
@ckipp01 Just a heads up that I've gone back to `2.12.10` on master so that I can keep working - no way I'm going back to Intellij! :-)
I've tried lots of different changes to the build to see if I can find the problem (plugins, sbt versions, compiler flags, wart remover, crossbuild/no cross build), but alas, no luck.
Thanks again for the assistance and work on the metals Vim integration - it's really great!

**Comment by ckipp01** [comment]
Alright, thanks for the instructions via Twitter. I was able to get everything pulled down, built etc. I then followed what you had listed above, and I do actually get the expected  error.

<img width="432" alt="Screenshot 2020-02-10 at 09 50 03" src="https://user-images.githubusercontent.com/13974112/74134509-fb5fa000-4bea-11ea-8ee5-25a4866fce16.png">

**Comment by tgodzik** [comment]
You might try to install Bloop manually and then run compile via CLI. By default Metals doesn't ship with the CLI. It has an additional `--verbose` option which might shed the light on the issue you are having.

Docs here: https://scalacenter.github.io/bloop/setup

**Comment by ckipp01** [comment]
Again, sorry for the confusion on this. I didn't checkout the 2.13 commit that you mentioned @davesmith00000. However, after checking it out, I am able to get the same issue. It's a bit odd. Even when I try to compile with bloop cli, I get the following.

```
❯ bloop compile indigoJVM
Deduplicating compilation of facadesJVM from bsp client 'Metals 0.8.0+203-04ec6bff-SNAPSHOT' (since 1m 44.739s)
Deduplicating compilation of sharedJVM from bsp client 'Metals 0.8.0+203-04ec6bff-SNAPSHOT' (since 20m 48.433s)
```
I get the same in my `.metals/metals.log` when I change something in that package and attempt to save. Any ideas @tgodzik?

**Comment by tgodzik** [comment]
@ckipp01 maybe try to increase ` -Xss` ? And what happens if you try to compile from CLI?

**Comment by davesmith00000** [comment]
I'm using a `.jvmopts` file to set the following:
```
-Xss8m
-Xms1G
-Xmx8G
```
Look reasonable?

**Comment by tgodzik** [comment]
@davesmith00000 It does seem alright, did you have it set before too and in the Bloop directory?
It's important to note that Bloop doesn't pick up `.jvmopts` from local workspaces, because a single Bloop server works for all Metals workspaces. You need to set them in `~/.bloop/.jvmopts`

**Comment by davesmith00000** [comment]
Ok I've been on an exciting tour. I installed bloop via homebrew and that doesn't seem to pick up jvmopts from `~/.bloop/.jvmopts` - luckily it will pick it up from the installation directory.

So I now how this:
```
> bloop server
Defaulting on nailgun port 8212
There is no server running at port 8212
Starting the bloop server... this may take a few seconds
Running /usr/local/Cellar/bloop/1.4.0-RC1/bin/blp-server as a jar...
Shelling out with '['java', '-Xss8m', '-Xms1G', '-Xmx8G', '-jar', '/usr/local/Cellar/bloop/1.4.0-RC1/bin/blp-server']' ...
Error: Invalid or corrupt jarfile /usr/local/Cellar/bloop/1.4.0-RC1/bin/blp-server
Running /usr/local/Cellar/bloop/1.4.0-RC1/bin/blp-server as a script...
Shelling out in Unix system with ['sh', '/usr/local/Cellar/bloop/1.4.0-RC1/bin/blp-server', '-J-Xss8m', '-J-Xms1G', '-J-Xmx8G']
Detected java options are '-Xss8m -Xms1G -Xmx8G'
Unable to load nailgun-version.properties.
NGServer [UNKNOWN] started on address localhost/127.0.0.1 port 8212
```

Slightly concerned about `Error: Invalid or corrupt jarfile /usr/local/Cellar/bloop/1.4.0-RC1/bin/blp-server` but that doesn't seem to stop it running...

Compilation against 2.13.1 still does nothing, 2.12.10 works perfectly.

**Comment by tgodzik** [comment]
@ckipp01 Did you manage to find anything? Or do you need a hand?

**Comment by ckipp01** [comment]
> @ckipp01 Did you manage to find anything? Or do you need a hand?

Hey! No, sorry, I haven't been able to look at this at all the last couple days! Sorry @davesmith00000! I'd say if you can @tgodzik, please do take a look.

**Comment by davesmith00000** [comment]
No need to apologise @ckipp01 - I'm grateful you've had time to look at all and validate that I'm not just going crazy! :-)

In any case, I've just now found the problem!!! ...I'm not claiming I understand it.

I thought I'd checked this before weirdly, but it's got something to do with the silencer compiler plugin.

Broken:
```
lazy val commonSettings = Seq(
  version := indigoVersion,
  scalaVersion := scala2,
  organization := "indigo",
  libraryDependencies ++= Seq(
    "com.lihaoyi" %%% "utest" % "0.6.9" % "test"
  ),
  libraryDependencies ++= Seq(
    compilerPlugin("com.github.ghik" % "silencer-plugin" % silencerVersion cross CrossVersion.full),
    "com.github.ghik" % "silencer-lib" % silencerVersion % Provided cross CrossVersion.full
  ),
  testFrameworks += new TestFramework("utest.runner.Framework"),
  scalacOptions in (Compile, doc) ++= Seq("-groups", "-implicits"),
  scalacOptions in (Compile, compile) ++= Scalac213Options.scala213Compile,
  scalacOptions in (Test, test) ++= Scalac213Options.scala213Test,
  wartremoverWarnings in (Compile, compile) ++= Warts.allBut(
    Wart.Overloading,
    Wart.ImplicitParameter
  ),
  scalacOptions += "-Yrangepos"
)
```

Fixed:
```
lazy val commonSettings = Seq(
  version := indigoVersion,
  scalaVersion := scala2,
  organization := "indigo",
  libraryDependencies ++= Seq(
    "com.lihaoyi" %%% "utest" % "0.6.9" % "test"
  ),
  // libraryDependencies ++= Seq(
  //   compilerPlugin("com.github.ghik" % "silencer-plugin" % silencerVersion cross CrossVersion.full),
  //   "com.github.ghik" % "silencer-lib" % silencerVersion % Provided cross CrossVersion.full
  // ),
  testFrameworks += new TestFramework("utest.runner.Framework"),
  scalacOptions in (Compile, doc) ++= Seq("-groups", "-implicits"),
  scalacOptions in (Compile, compile) ++= Scalac213Options.scala213Compile,
  scalacOptions in (Test, test) ++= Scalac213Options.scala213Test,
  wartremoverWarnings in (Compile, compile) ++= Warts.allBut(
    Wart.Overloading,
    Wart.ImplicitParameter
  ),
  scalacOptions += "-Yrangepos"
)
```

**Comment by davesmith00000** [comment]
I'm only actually using the plugin on one sub module, and I can get rid of it because the code it's attached too was just a quick test anyway, so that should unblock me. It would be interesting to know what's going on though, and why it's only affecting that specific version of Scala.

**Comment by davesmith00000** [comment]
I'm going to close this issue on the basis that we have a solution, and I'll raise an issue with the silencer project: https://github.com/ghik/silencer

Thanks again for the assistance.

**Comment by davesmith00000** [comment]
In fact there already is one! Ah well...
https://github.com/ghik/silencer/issues/43

**Comment by ckipp01** [comment]
Glad you were able to get it figured out @davesmith00000 👍

**Comment by timcharper** [comment]
I just ran in to this as well, sure enough, silencer plugin + semanticdb plugin = problems.

**Comment by tgodzik** [comment]
@timcharper This should be fixed in latest scalameta 4.3.7, it will soon be included in Metals SNAPSHOT versions.
